### PR TITLE
AAKD, on Python SDK, when want to deprecate arguments, I have the appropriate tooling

### PR DESCRIPTION
**Pull request by theodu**
- [ ] I opened a merge request on `kili` on a branch with the same name than the branch of the current pull request and forced security tests.
- [ ] I opened a pull request on the Node playground if the changes are breaking.

### CODE REVIEW

File: kili/helpers.py [hunk 1]
```diff
@@ -215,18 +216,34 @@ def fragment_builder(fields, type_of_fields):
     return fragment
 
 
-def deprecate(msg, _type=DeprecationWarning):
+def deprecate(
+        msg: Optional[str] = None,
+        removed_in: Optional[str] = None,
+        _type=DeprecationWarning):
     """
```
Comment (p-desaintchamas) [inline kili/helpers.py:223]: You should add a `use` to describe the case where you can use it, better than any description

**Review by p-desaintchamas — APPROVED** [review]

File: kili/helpers.py [hunk 2]
```diff
@@ -215,18 +216,34 @@ def fragment_builder(fields, type_of_fields):
...
+            in which the deprecation element has to be removed
         type: DeprecationWarning by default
     """
     def decorator(func):
-        @functools.wraps(func)
+        if removed_in:
+            if len(removed_in.split('.')) != 2:
+                raise ValueError(
+                    f'"removed_in" argument in deprecate wrapper of the function {func.__name__}'
+                    'should have the format "Major.Minor"')
```
Comment (crsegerie) [inline kili/helpers.py:240]: Major.Minor ? or {Major}.{Minor}

File: kili/helpers.py [hunk 3]
```diff
@@ -215,18 +216,34 @@ def fragment_builder(fields, type_of_fields):
...
 
 
-def deprecate(msg, _type=DeprecationWarning):
+def deprecate(
+        msg: Optional[str] = None,
+        removed_in: Optional[str] = None,
+        _type=DeprecationWarning):
     """
-    Decorator that generates a deprecation message
+    Decorator that tag a deprecated function.
```
Comment (crsegerie) [inline kili/helpers.py:224]: It's not a decorator, It's a decorator factory (higher level decorator)

File: kili/helpers.py [hunk 4]
```diff
@@ -215,18 +216,34 @@ def fragment_builder(fields, type_of_fields):
...
+    but still tag the function with this decorator, without giving a message argument
 
     Args:
-        msg: string message
+        msg: string message that will be displayed whenever the function is called
+        removed_in: string version in the format "Major.Minor"
+            in which the deprecation element has to be removed
         type: DeprecationWarning by default
     """
     def decorator(func):
```
Comment (crsegerie) [inline kili/helpers.py:235]: I think this decorator will obscure the type hint of the functions.

Comment (crsegerie) [reply kili/helpers.py:235]: The decorator can be typed with TypeVar  to enable type hints. https://stackoverflow.com/questions/54964760/is-this-decorator-type-annotated-correctly.

**Review by crsegerie — CHANGES_REQUESTED** [review]
If we want to enable type hints for the SDK in the future, we should type the decorator accordingly.

File: test/test_deprecation.py [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+import inspect
+from kili import client
+from kili import __version__
+
+
+def test_catch_deprecated_elements_to_remove():
+    """Raise an error if a function has a deprecation tag of a version
+    that is bigger that the version that is released."""
+    functions = inspect.getmembers(client.Kili, inspect.isfunction)
```
Comment (crsegerie) [inline test/test_deprecation.py:11]: Super cool!

File: kili/helpers.py (hunk 2 above)
Comment (theodu) [reply kili/helpers.py:240]: Major.Minor, it is just a generic message

**Review by crsegerie — APPROVED** [review]

**Pull request merged**